clients/src/main/resources/common/message/BeginQuorumEpochRequest.json

```diff
@@ -18,24 +18,38 @@
   "type": "request",
   "listeners": ["controller"],
   "name": "BeginQuorumEpochRequest",
-  "validVersions": "0",
-  "flexibleVersions": "none",
+  // Version 1 adds flexible versions, voter key and leader endpoints (KIP-853)
+  "validVersions": "0-1",
+  "flexibleVersions": "1+",
   "fields": [
     { "name": "ClusterId", "type": "string", "versions": "0+",
       "nullableVersions": "0+", "default": "null"},
+    { "name": "VoterId", "type": "int32", "versions": "1+", "entityType": "brokerId", "ignorable": true,
+      "about": "The voter ID of the receiving replica" },
     { "name": "Topics", "type": "[]TopicData",
       "versions": "0+", "fields": [
-      { "name": "TopicName", "type": "string", "versions": "0+", "entityType": "topicName",
-        "about": "The topic name." },
-      { "name": "Partitions", "type": "[]PartitionData",
-        "versions": "0+", "fields": [
-        { "name": "PartitionIndex", "type": "int32", "versions": "0+",
-          "about": "The partition index." },
-        { "name": "LeaderId", "type": "int32", "versions": "0+", "entityType": "brokerId",
-          "about": "The ID of the newly elected leader"},
-        { "name": "LeaderEpoch", "type": "int32", "versions": "0+",
-          "about": "The epoch of the newly elected leader"}
-      ]}
-    ]}
+        { "name": "TopicName", "type": "string", "versions": "0+", "entityType": "topicName",
+          "about": "The topic name" },
+        { "name": "Partitions", "type": "[]PartitionData",
+          "versions": "0+", "fields": [
+            { "name": "PartitionIndex", "type": "int32", "versions": "0+",
+              "about": "The partition index" },
+            { "name": "VoterDirectoryId", "type": "uuid", "versions": "1+", "ignorable": true,
+              "about": "The directory id of the receiving replica" },
+            { "name": "LeaderId", "type": "int32", "versions": "0+", "entityType": "brokerId",
+              "about": "The ID of the newly elected leader"},
+            { "name": "LeaderEpoch", "type": "int32", "versions": "0+",
+              "about": "The epoch of the newly elected leader"}
+          ]
+        }
+      ]
+    },
+    { "name": "LeaderEndpoints", "type": "[]LeaderEndpoint", "versions": "1+", "ignorable": true,
+      "about": "Endpoints for the leader", "fields": [
+        { "name": "Name", "type": "string", "versions": "1+", "mapKey": true, "about": "The name of the endpoint" },
+        { "name": "Host", "type": "string", "versions": "1+", "about": "The node's hostname" },
+        { "name": "Port", "type": "int32", "versions": "1+", "about": "The node's port" }
```

Comment:
Why is this `int32` rather than `int16`? Ports can't be greater than 2**16, right?

Reply:
Good catch. I updated all of the affect RPCs to use `unit16` for the ports.